src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointEvidence.java

```diff
@@ -0,0 +1,476 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.apache.spark.serializer.KryoRegistrator;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Various types of read anomalies that provide evidence of genomic breakpoints.
+ */
+public class BreakpointEvidence implements Comparable<BreakpointEvidence> {
+    private final short contigIndex;
+    private final short eventWidth;
+    private final int contigStart;
+    private final String templateName;
+    private final TemplateEnd templateEnd;
+
+    public enum TemplateEnd {
+        UNPAIRED(""), PAIRED_UNKNOWN("/?"), PAIRED_FIRST("/1"), PAIRED_SECOND("/2"), PAIRED_INTERIOR("/0");
```

Comment:
Defensive.  A read that's flagged neither first nor last must be somewhere in the interior of the template.  SAM spec makes it possible.